server/docs/design/block-persistence.md

```diff
@@ -3,64 +3,213 @@
 ## Table of Contents
 
 1. [Purpose](#purpose)
+1. [Terminology](#terminology)
+1. [Abstractions](#abstractions)
 1. [Goals](#goals)
-1. [Terms](#terms)
-1. [Entities](#entities)
-1. [Design](#design)
-
----
+1. [Overview](#overview)
+1. [Implementations](#implementations)
+   1. [Block as Local Directory (a.k.a. `block-as-local-dir`)](#block-as-local-directory-aka-block-as-local-dir)
+      1. [Purpose](#purpose-1)
+      1. [Overview](#overview-1)
+      1. [Configurable Parameters](#configurable-parameters)
+   1. [Block as Local File (a.k.a. `block-as-local-file`)](#block-as-local-file-aka-block-as-local-file)
+      1. [Purpose](#purpose-2)
+      1. [Overview](#overview-2)
+      1. [Trie Structure & Algorithm for Block Path Resolution](#trie-structure--algorithm-for-block-path-resolution)
+      1. [Configurable Parameters](#configurable-parameters-1)
+   1. [No Operation (a.k.a. `no-op`)](#no-operation-aka-no-op)
+      1. [Purpose](#purpose-3)
+      1. [Overview](#overview-3)
 
 ## Purpose
 
-The main objective of the `hedera-block-node` project is to replace the storage of Consensus Node artifacts (e.g.
-Blocks) on cloud storage buckets (e.g. GCS and S3) with a solution managed by the Block Node server. This document aims
-to describe the high-level design of how the Block Node persists and retrieves Blocks and how it handles exception cases
-when they arise.
+A major objective of the `hedera-block-node` project is to replace the storage
+of Consensus Node artifacts (e.g. Blocks) on cloud storage buckets (e.g. GCS and
+S3) with a solution managed by the Block Node server. This document aims to
+describe the high-level design of how the Block Node persists and retrieves
+Blocks and how it handles exception cases when they arise.
 
----
+## Terminology
 
-## Goals
+<dl>
+  <dt>BlockItems</dt>
+  <dd>The `Block` description and protobuf definition are maintained in the
+  `hedera-protobuf` <a href="https://github.com/hashgraph/hedera-protobufs/blob/continue-block-node/documents/api/block/stream/block.md">project</a>.</dd>
 
-1) BlockItems streamed from a producer (e.g. Consensus Node) must be collated and persisted as a Block.  Per the
-   specification, a Block is an ordered list of BlockItems. How the Block is persisted is an implementation detail.
-2) A Block must be efficiently retrieved by block number.
+  <dt>BlockItemUnparsed</dt>
+  <dd>An unparsed, raw version of a `BlockItem`.</dd>
 
----
+  <dt>BlockUnparsed</dt>
+  <dd>An unparsed, raw version of a `Block`.</dd>
 
-## Terms
+  <dt>BlockNumber</dt>
+  <dd>A value that represents the unique number (identifier) of a given `Block`.
+  It is an auto-incrementing `long` value, starting from `0` (zero).</dd>
```

Comment:
Slightly incorrect description. The block number is not automatic increment; we must read it from each block header, and it might (in very rare situations) skip a number.
Also, English convention is to spell the number first, and place the digit in parenthetical (if included).
```suggestion
  It is a strictly increasing `long` value, starting from zero (`0`).</dd>
```

Reply:
✅ Done.